intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/detector/query_explainer.py

```diff
@@ -0,0 +1,39 @@
+# !/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Polish and clarify the input user query with LLM."""
+
+from intel_extension_for_transformers.neural_chat.pipeline.plugins.prompt.prompt_template \
+     import polish_query_prompt
+from intel_extension_for_transformers.neural_chat.models.model_utils import predict
+
+
+class QueryPolisher:
+    def __init__(self):
+        pass
+
+    def polish_query(self, model_name, query):
+        """Using the LLM to polish the user query to fix the gramma errors and clarify the user query."""
+        prompt = polish_query_prompt(query)
+        params = {}
+        params["model_name"] = model_name
+        params["prompt"] = prompt
+        params["temperature"] = 0.1
+        params["top_k"] = 3
+        params["max_new_tokens"] = 512

```

Comment:
why limit to 512 for max length, and similar to temperature/top_k?

Reply:
this is only use for user query, which could not be too long. So, i set 512 for a considerable length tolerance. temperature value and top-k value is verified effective on the local machine. There is no general guideline for these settings. select the same way as intent detection to set these values.

Reply:
the query should support up to the max LLM context length? what's the side effect, if the query is longer than 512?

Reply:
in general, the user query should be less than 512 token length. we do not expect the user to change this parameter in RAG application. if there is some special use case, we can change this value